### DISCUSSION TITLE
Help us with a tutorial for normal internet users?

### DISCUSSION THREAD

**Discussion by siavashi** [apernet/hysteria#455] [Q&A]
Hello, Can you kindly help normal users like me with an easy step-by-step tutorial on setting up Hysteria on a VPS?

Due to the internet crackdown in IRAN, almost none of the prior methods of evading censorship are working. An easy-to-follow tutorial would be greatly appreciated and can help many people.

Thanks

**Comment by tobyxdd** [comment]
Hi siavashi,

We already have this quick start page in wiki: https://github.com/HyNetwork/hysteria/wiki/Quick-Start

Although this assumes you already have a basic knowledge of Linux and how to use a proxy. As a "normal internet user", what areas do you think our current documentation is lacking? We are happy to improve it based on your feedback.

**Reply by siavashi** [reply]
Hi, Thanks for the prompt reply. Unfortunately, I don't know anything about Linux.
The main problem for me is that I didn't know where to begin, like when I want to get a VPS, like what Operating System and which version of it, I should get for setting up Hysteria. CentOS, Debian,Fedora, Ubuntu or AlmaLinux ? A page like this gives lots of choices for operating systems  :
https://alphavps.com/clients/store/ssd-openvz-virtual-servers
And I don't know where to begin after getting the VPS. for example the very first sentence of the Quickstart begins with:
"Create a config.json under the root directory of the program:"
Does it mean that I have to unzip the hysteria file in the root directory first? ("the program" is the Hysteria folder, right?)

I know I am asking too much, but would it be possible to add a screenshot of every step of the way from the very beginning? or just make a video? so that we can simply follow it.

Thanks

**Reply by tobyxdd** [reply]
You can use Hysteria on pretty much any VPS and Linux distribution (although for beginners I recommend Ubuntu).

> Does it mean that I have to unzip the hysteria file in the root directory first

Unzip it anywhere you want, put `config.json` along with it.

There are already some bloggers who have written beginner tutorials, but afaik they are all in Chinese. You did make a good point that we could have official beginner tutorials/tools for wider adoption.

**Comment by mostafaznv** [comment]
I tried this in Iran. It doesn't work.
I think it's because of UDP restrictions in Iran.

Is there any chance to use faketcp without root permission? so we can use it in non-linux clients (iOS, windows ...)

**Reply by tobyxdd** [reply]
Is UDP completely blocked in Iran?

**Reply by siavashi** [reply]
- I am using a free server that has Hysteria installed and it's working pretty well (Also in IRAN).

-I still manage to connect with 1-2 OpenVPN UDP servers that haven't been blocked yet. So I don't UDP is blocked, but generally speaking, what I noticed is that a solution that works when using a connection from Operator or ISP A, doesn't necessarily work on Operator/ISP B. 

I also found NaiveProxy to be an effective solution in Iran. Unlike Hysteria, there are some English tutorials out there.

**Comment by Recoba86** [comment]
use this Tutorial
its in English

[https://www.oilandfish.com/posts/hynetwork-hysteria.html](url)

**Comment by mega-optimus** [comment]
You can find any popular tutorial, translate it with google translate.
